### PR TITLE
feat: add message properties and event definition

### LINKED ISSUE #96: Support message templates in JSON schema
__What should we do?__

Support templates like:

```json
{
  "name": "Message Template",
  "id": "com.camunda.example.MessageTemplate",
  "appliesTo": [
    "bpmn:IntermediateCatchEvent"
  ],
  "elementType": {
    "value": "bpmn:IntermediateCatchEvent",
    "eventDefinition": "bpmn:MessageEventDefinition"
  },
  "properties": [
      {
        "label": "Message name",
        "type": "String",
        "generatedValue": "uuid",
        "editable": false,
        "binding": {
          "type": "bpmn:Message#property",
          "name": "name"
        }
      },
      {
        "label": "Correlation Key",
        "type": "String",
        "feel": true,
        "binding": {
          "type": "bpmn:Message#zeebe:subscription#property",
          "name": "correlationKey"
        }
      }
  ]
}
```

__Why should we do it?__

Required for https://github.com/camunda/camunda-modeler/issues/3403

### PR DESCRIPTION
**Pull request by barmac**
Closes #96

### CODE REVIEW

File: packages/zeebe-element-templates-json-schema/src/schema.json [hunk 1]
```diff
@@ -18,6 +18,48 @@
...
+                  "eventDefinition": {
+                    "const": "bpmn:MessageEventDefinition"
+                  }
+                },
+                "required": [
+                  "eventDefinition"
+                ]
+              }
+            }
+          }
```
Comment (barmac) [inline packages/zeebe-element-templates-json-schema/src/schema.json:62]: This giant ensures that you put right `eventDefinition` in place once you try to add bpmn:Message bindings.

Comment (barmac) [reply packages/zeebe-element-templates-json-schema/src/schema.json:62]: If you have an idea how to simplify this, let me know!

Comment (smbea) [reply packages/zeebe-element-templates-json-schema/src/schema.json:62]: I'm going to have a little think if we can simplify this although at first glance I'm not sure. But could this be extracted to a sub-schema to not be so confusing on the main one?

Comment (smbea) [reply packages/zeebe-element-templates-json-schema/src/schema.json:62]: I think we do some simpler binding ifs/thens in for c7 that are `properties.json` https://github.com/camunda/element-templates-json-schema/blob/master/packages/element-templates-json-schema/src/defs/properties.json#L13

Comment (barmac) [reply packages/zeebe-element-templates-json-schema/src/schema.json:62]: Yeah we can extract it.

Comment (barmac) [reply packages/zeebe-element-templates-json-schema/src/schema.json:62]: Indeed, but here we have to define constraint on a property up the tree, i.e. the common parent is the template only :/

Comment (barmac) [reply packages/zeebe-element-templates-json-schema/src/schema.json]: OK I've extracted the schema to [template.json](packages/zeebe-element-templates-json-schema/src/defs/template.json)

Comment (smbea) [reply packages/zeebe-element-templates-json-schema/src/schema.json]: Uhm I see.

Do we need to double `properties` at the start of the `if`? I tested without it on a validator and it seemed to work
```diff
"if": {
-  "properties": {
       "properties": {
           "contains": {
```

Comment (barmac) [reply packages/zeebe-element-templates-json-schema/src/schema.json]: If I remove one of the properties, the validation does not run properly. Note that one `properties` is a JSON schema keyword while another is template's property called `properties`.

Comment (barmac) [reply packages/zeebe-element-templates-json-schema/src/schema.json]: So it reads `If property "properties" contains an item...` opposed to `If property "contains"...`

Comment (smbea) [reply packages/zeebe-element-templates-json-schema/src/schema.json]: Ah yes sorry about that. I must have tested the wrong generated schema. Makes sense. I think it's ok as is then. I'll check out the rest of the PR

**Review by smbea — APPROVED** [review]

**Pull request merged**